src/OmniSharp.Roslyn.CSharp/Services/Signatures/CheckStaticInvocation.cs

```diff
@@ -0,0 +1,175 @@
+                        return ((ConversionOperatorDeclarationSyntax)member).Modifiers;
+                    case SyntaxKind.MethodDeclaration:
+                        return ((MethodDeclarationSyntax)member).Modifiers;
+                    case SyntaxKind.GetAccessorDeclaration:
+                    case SyntaxKind.SetAccessorDeclaration:
+                    case SyntaxKind.AddAccessorDeclaration:
+                    case SyntaxKind.RemoveAccessorDeclaration:
+                        return ((AccessorDeclarationSyntax)member).Modifiers;
+                }
+            }
+
+            return default;
+        }
+        public static bool IsFoundUnder<TParent>(this SyntaxNode node, Func<TParent, SyntaxNode> childGetter)
+           where TParent : SyntaxNode
+        {
+            var ancestor = node.GetAncestor<TParent>();
+            if (ancestor == null)
+            {
+                return false;
+            }
+
+            var child = childGetter(ancestor);
+
+            // See if node passes through child on the way up to ancestor.
+            return node.GetAncestorsOrThis<SyntaxNode>().Contains(child);
+        }
+        public static TNode GetAncestor<TNode>(this SyntaxNode node)
+           where TNode : SyntaxNode
+        {
+            var current = node.Parent;
+            while (current != null)
+            {
+                if (current is TNode tNode)
+                {
+                    return tNode;
+                }
+
+                current = current.GetParent();
+            }
+
+            return null;
+        }
+        private static SyntaxNode GetParent(this SyntaxNode node)
+        {
+            return node is IStructuredTriviaSyntax trivia ? trivia.ParentTrivia.Token.Parent : node.Parent;
+        }
+
+        public static TNode FirstAncestorOrSelfUntil<TNode>(this SyntaxNode node, Func<SyntaxNode, bool> predicate)
+            where TNode : SyntaxNode
+        {
+            for (var current = node; current != null; current = current.GetParent())
+            {
+                if (current is TNode tnode)
+                {
+                    return tnode;
+                }
+
+                if (predicate(current))
+                {
+                    break;
+                }
+            }
+
+            return default;
+        }
+
+        public static TNode GetAncestorOrThis<TNode>(this SyntaxNode node)
+                where TNode : SyntaxNode
+        {
+            return node?.GetAncestorsOrThis<TNode>().FirstOrDefault();
+        }
+
+        public static IEnumerable<TNode> GetAncestorsOrThis<TNode>(this SyntaxNode node)
+            where TNode : SyntaxNode
+        {
+            var current = node;
+            while (current != null)
+            {
+                if (current is TNode tNode)
+                {
+                    yield return tNode;
+                }
+
+                current = current.GetParent();
+            }
+        }
+    }
+}
+
```

Comment:
Is this a bunch of code copied from Roslyn? If so, this should be commented per the MIT license in the Roslyn code base. It might also be worth adding a TODO comment to remove this once we've added a proper public API for Signature Help to Roslyn.